### PR TITLE
Use authorizer v2

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

**Review by ronenh — APPROVED** [review]
Looks great. You probably want to make a corresponding change in [aserto-docs](https://github.com/aserto-dev/aserto-docs/tree/main/docs/software-development-kits/go)

File: README.md [hunk 1]
```diff
@@ -89,13 +89,13 @@ Use the client's `Is()` method to request authorization decisions from the Asert
...
-	authz "github.com/aserto-dev/go-grpc-authz/aserto/authorizer/authorizer/v1"
-	"github.com/aserto-dev/go-grpc/aserto/api/v1"
+	authz "github.com/aserto-dev/go-authorizer/aserto/authorizer/v2"
+	"github.com/aserto-dev/go-authorizer/aserto/authorizer/v2/api"
 )
 
 resp, err := authorizer.Is(c.Context, &authz.IsRequest{
 	PolicyContext: &api.PolicyContext{
-		Id:        "peoplefinder",
+		Name:      "peoplefinder",
```
Comment (carabasdaniel) [inline README.md:98]: The name is not needed, either replace the id by the pair of name and label (instance label) or just don't add it

File: examples/middleware/http/gorilla/main.go [hunk 2]
```diff
@@ -29,7 +29,7 @@ func main() {
 	mw := std.New(
 		authClient,
 		middleware.Policy{
-			ID:       "local",
+			Name:     "local",
```
Comment (carabasdaniel) [inline examples/middleware/http/gorilla/main.go:32]: see previous comments

File: middleware/grpc/internal/pbutil/select_test.go [hunk 3]
```diff
@@ -7,15 +7,15 @@ import (
...
+	"github.com/aserto-dev/go-authorizer/aserto/authorizer/v2"
+	"github.com/aserto-dev/go-authorizer/aserto/authorizer/v2/api"
 	"google.golang.org/protobuf/types/known/fieldmaskpb"
 )
 
 func TestFieldMaskIsValid(t *testing.T) {
 	msg := &authorizer.IsRequest{
 		PolicyContext: &api.PolicyContext{
-			Id:   "policyID",
+			Name: "policyName",
```
Comment (carabasdaniel) [inline middleware/grpc/internal/pbutil/select_test.go:18]: see previous comments

File: middleware/grpc/internal/pbutil/select_test.go [hunk 4]
```diff
@@ -73,7 +73,7 @@ func TestFieldMaskIsValid(t *testing.T) {
 		[]string{"policy_context"},
 		map[string]interface{}{
 			"policy_context": map[string]interface{}{
-				"id":   "policyID",
+				"name": "policyName",
```
Comment (carabasdaniel) [inline middleware/grpc/internal/pbutil/select_test.go:76]: see previous comments

File: middleware/http/std/example_test.go [hunk 5]
```diff
@@ -33,7 +33,7 @@ func Example() {
 	middleware := mw.New(
 		authorizer,
 		mw.Policy{
-			ID:       "<Aserto policy ID>",
+			Name:     "<Aserto policy Name>",
```
Comment (carabasdaniel) [inline middleware/http/std/example_test.go:36]: see previous comments

File: middleware/policy.go [hunk 6]
```diff
@@ -9,8 +9,8 @@ package middleware
 
 // Policy holds authorization options that apply to all requests.
 type Policy struct {
-	// ID is the ID of the aserto policy being queried for authorization.
-	ID string
+	// Name is the Name of the aserto policy being queried for authorization.
+	Name string
```
Comment (carabasdaniel) [inline middleware/policy.go:13]: should contain the instance label also

**Review by carabasdaniel — CHANGES_REQUESTED** [review]
The changes do not replace the policy.id with policy name in v2

File: README.md [hunk 7]
```diff
@@ -221,7 +223,7 @@ import (
 middleware, err := grpcmw.New(
 	client,
 	middleware.Policy{
-		ID: "<Policy ID>",
+		Name: "<Policy Name>",
```
Comment (carabasdaniel) [inline README.md:226]: Do you not need the instance label in the middleware.Policy declaration ?

**Review by carabasdaniel — APPROVED** [review]

**Pull request merged**